.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,69 @@
+name: Java CI with Maven and Release Artifact
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  release:
+    types: [ released ]
+
+permissions:
+  contents: write
+  actions: read
+  checks: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    outputs:
+      version: ${{ steps.extract_version.outputs.version }}
+      artifactId: ${{ steps.extract_artifactId.outputs.artifactId }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '23'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Build with Maven
+        run: mvn -B clean install
+
+      - name: Extract project version
+        id: extract_version
+        run: |
+          VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Extract artifactID
+        id: extract_artifactId
+        run: |
+          ARTIFACT_ID=$(mvn help:evaluate -Dexpression=project.artifactId -q -DforceStdout)
+          echo "artifactId=$ARTIFACT_ID" >> $GITHUB_OUTPUT
+
+      - name: Upload JAR
```

Comment:
Just to understand: do we need to do this upload JAR so that we can later download it again for the release? Or may it be sufficient to only do the upload in case of a release?